tests/test_fit_result.py

```diff
@@ -155,28 +114,21 @@ def test_fitting_2(self):
         fit = Fit(model, xx, yy, ydata)
         fit_result = fit.execute()
 
-        for param in fit_result.params:
+        for param in fit.model.params:
             self.assertAlmostEqual(fit_result.stdev(param)**2, fit_result.variance(param))
-            self.assertEqual(fit_result.stdev(param), fit_result.params.stdev(param))
-            self.assertEqual(fit_result.value(param), fit_result.params.value(param))
 
         # Covariance matrix should be symmetric
-        for param_1 in fit_result.params:
-            for param_2 in fit_result.params:
+        for param_1 in fit.model.params:
+            for param_2 in fit.model.params:
                 self.assertAlmostEqual(fit_result.covariance(param_1, param_2), fit_result.covariance(param_2, param_1))
-#        print(fit_result.params.covariance_matrix)
-#        print(fit_result.covariance(x0_1, x0_2))
-
-        with warnings.catch_warnings(record=True) as w:
-            # Cause all warnings to always be triggered.
-            warnings.simplefilter("always")
-            # Trigger DeprecationWarning
-            fit_result.params.get_stdev(x0_1)
-            fit_result.params.get_value(x0_1)
-            self.assertTrue(len(w) == 2)
-            for warning in w:
-                self.assertTrue(issubclass(warning.category, DeprecationWarning))
 
+    def test_minimizer_included(self):
+        """"The minimizer used should be included in the results."""
+        return NotImplementedError()
+
+    def test_objective_included(self):
+        """"The objective used should be included in the results."""
+        return NotImplementedError()
```

Comment:
Excellent ideas. That way you could even conceive letting the minimizer (or objective?) keep track of the number of evaluations. Let's leave it for later though.